psi4/src/psi4/libfock/snLinK.cc

```diff
@@ -0,0 +1,538 @@
+    integrator_settings_.screen_ek = (do_no_screen || cutoff_ == 0.0) ? false : true;
+
+    auto ints_tol = options.get_double("SNLINK_INTS_TOLERANCE");
+    integrator_settings_.energy_tol = ints_tol; 
+    integrator_settings_.k_tol = ints_tol;
+
+    // set other integrator options
+    const std::string integrator_input_type = "Replicated";
+    const std::string integrator_kernel = options_.get_str("SNLINK_INTEGRATOR_KERNEL");  
+    const std::string reduction_kernel = options_.get_str("SNLINK_REDUCTION_KERNEL");  
+    const std::string lwd_kernel = options_.get_str("SNLINK_LWD_KERNEL");  
+
+    // construct integrator factory 
+    integrator_factory_ = std::make_unique<GauXC::XCIntegratorFactory<matrix_type> >(ex, integrator_input_type, integrator_kernel, lwd_kernel, reduction_kernel);
+ 
+    timer_off("snLinK: Construct GauXC Factories");
+    
+    // => Part #3: Construct GauXC Integrator <= //
+    timer_on("snLinK: Construct GauXC Integrator");
+
+    radial_points_ = options_.get_int("SNLINK_RADIAL_POINTS");
+    spherical_points_ = options_.get_int("SNLINK_SPHERICAL_POINTS");
+
+    // convert Psi4 fundamental quantities to GauXC 
+    auto gauxc_mol = psi4_to_gauxc_molecule(primary_->molecule());
+    auto gauxc_primary = psi4_to_gauxc_basisset<double>(primary_, basis_tol_, force_cartesian);
+
+    // create snLinK grid for GauXC
+    auto grid_batch_size = options_.get_int("SNLINK_GRID_BATCH_SIZE");
+    auto use_debug_grid = options_.get_bool("SNLINK_USE_DEBUG_GRID");
+
+    auto gauxc_grid = !use_debug_grid ? GauXC::MolGridFactory::create_default_molgrid(
+        gauxc_mol, 
+        pruning_scheme_map[pruning_scheme_],
+        GauXC::BatchSize(grid_batch_size), 
+        radial_scheme_map[radial_scheme_], 
+        GauXC::RadialSize(radial_points_),
+        GauXC::AngularSize(spherical_points_)
+    ) :
+    GauXC::MolGridFactory::create_default_molgrid(
+        gauxc_mol, 
+        pruning_scheme_map[pruning_scheme_],
+        GauXC::BatchSize(grid_batch_size), 
+        radial_scheme_map[radial_scheme_], 
+        GauXC::AtomicGridSizeDefault::UltraFineGrid
+    );
+   
+    // construct load balancer
+    //const size_t quad_pad_value = 1;
+    //auto gauxc_load_balancer = gauxc_load_balancer_factory_->get_instance(*rt, gauxc_mol, gauxc_grid, gauxc_primary, quad_pad_value);
+    auto gauxc_load_balancer = gauxc_load_balancer_factory_->get_instance(*rt, gauxc_mol, gauxc_grid, gauxc_primary);
+    
+    // construct weights module
+    auto gauxc_mol_weights = gauxc_mol_weights_factory_->get_instance();
+
+    // apply partition weights
+    gauxc_mol_weights.modify_weights(gauxc_load_balancer);
+
+    // construct dummy functional
+    // note that the snLinK used by the eventually-called eval_exx function is agnostic to the input functional!
+    const std::string dummy_func_str = "B3LYP";
+    auto spin = options_.get_str("REFERENCE") == "UHF" ? ExchCXX::Spin::Polarized : ExchCXX::Spin::Unpolarized;
```

Comment:
Hmm, what about ROHF?

Reply:
Fixed! As per online discussion elsewhere, this won't have any practical effects on the end results since the functional is just a dummy functional in this context, but it's good to have this corrected for clarity purposes anyway.